Include new_charged_registrations in "New registrations" search results

PaulDoyle-EA opened the pull request:
The dashboard search option for "New registrations" did not previously return any private beta transient registrations. This change searches on a subset of transient_registration classes so that both new_registrations and new_charged_registrations (but no other transient_registration types) will be returned in the search results.
https://eaflood.atlassian.net/browse/RUBY-3579

PaulDoyle-EA committed:
Include new_charged_registrations in "New registrations" search results

PaulDoyle-EA committed:
Merge branch 'main' into feature/RUBY-3579_find_pb_in_progress_regs

PaulDoyle-EA committed:
Remove duplicate search code

brujeo committed:
RUBY-3554: Manage Private Beta participants page (#1603)

* Add BetaParticipantsController and related views

This commit adds a new controller, `BetaParticipantsController`, along with its corresponding views. The controller includes actions for index, new, and create. The index action retrieves all `WasteExemptionsEngine::BetaParticipant` records and orders them by `created_at` in descending order. The new action initializes a new `WasteExemptionsEngine::BetaParticipant` object. The create action creates a new `WasteExemptionsEngine::BetaParticipant` record based on the submitted parameters. If the record is successfully saved, the user is redirected to the index page; otherwise, the new page is rendered again.

The views include an index page that displays a list of participants, with their registration number, email, registration details, and a link to a private beta page. The index page also includes a link to add a new participant. The new page contains a form to add a new participant, with fields for registration number and email.

This commit also adds localization files for the beta participants feature, including English translations for the headings, labels, and error messages.

Routes are updated to include the routes for the `BetaParticipantsController`, allowing access to the index, new, and create actions.

* [RUBY-3554] Refactoring private beta participants page

* [RUBY-3554] Adding a link to private beta participants page

* [RUBY-3554] Refactor beta participant factory

* [RUBY-3554] Refactor helper spec to use NewRegistrationsHelper instead of RegistrationsHelper

* [RUBY-3554] Refactor beta_participants index view

* [RUBY-3554] Refactor beta_participants index view and add handling for no results

* [RUBY-3554] Refactor beta_participants_spec and add test for no results

* [RUBY-3554] Refactor RegistrationsHelper and NewRegistrationsHelper

* [RUBY-3554] Refactor NewRegistrationsHelperSpec and remove trailing whitespace

* [RUBY-3554] Refactor NewRegistrationsHelperSpec

* [RUBY-3554] Refactor NewRegistrationsHelperSpec and remove trailing whitespace

* [RUBY-3554] Refactor application layout to show PB users page link regardless of feature toggle

* [RUBY-3554] Refactor RegistrationsHelperSpec and NewRegistrationsHelperSpec and move common tests into shared_examples

brujeo commented on app/models/concerns/can_be_searched_like_registration.rb:
```diff
@@ -3,8 +3,10 @@
 module CanBeSearchedLikeRegistration
   extend ActiveSupport::Concern
 
+  # rubocop:disable Metrics/BlockLength
```
I'm not a big fan of inline rubocop:disable comments. If it's a permanent thing, then better to move it to rubocop config and exclude this concern from Metrics/BlockLength checks

PaulDoyle-EA committed:
Fix/ruby 3559 3607 private beta resource flash (#1610)

https://eaflood.atlassian.net/browse/RUBY-3559
https://eaflood.atlassian.net/browse/RUBY-3607

PaulDoyle-EA committed:
Move cop exclusion from source file to rubocop config

PaulDoyle-EA committed:
Merge branch 'main' into feature/RUBY-3579_find_pb_in_progress_regs
